docs/advanced_topics.rst

```diff
@@ -1,114 +1,244 @@
-
+This example sets thread 0 to CPU 3, thread 1 to CPU 5, thread 2 to CPU 6, thread 3 to CPU 10,
+and thread 4 to the CPU ID that is returned by nvmlDeviceGetCpuAffinity.
 
-Memory consumption
+Memory Consumption
 ------------------
 
-DALI uses three kinds of memory: host, host page-locked (pinned) and GPU.
-GPU and host pinned memory allocation and freeing require device synchronization. For this reason, DALI avoids reallocating these kinds of memory whenever possible. The buffers allocated with this kind of storage will only grow when the existing buffer is too small to accommodate the requested shape. This allocation strategy reduces the number of total memory management operations and greatly increases the processing speed after the allocations have stabilized.
-
-In contrast, ordinary host memory is relatively cheap to allocate and free. To reduce host memory consumption, the buffers may shrink if the new requested size is smaller than a specified fraction of the old size (called shrink threshold). It can be adjusted to any value between 0 (never shrink) and 1 (always shrink), with the default being 0.9. The value can be controlled either via environment variable `DALI_HOST_BUFFER_SHRINK_THRESHOLD` or set in Python using `nvidia.dali.backend.SetHostBufferShrinkThreshold` function.
-
-During processing, DALI works on batches of samples. For GPU and some CPU Operators, the batch is stored as contiguous allocation which is processed in one go, which reduces the number of necessary allocations.
-For some CPU Operators, that cannot calculate their output size ahead of time, the batch is instead stored as a vector of separately allocated samples.
+DALI uses the following memory types:
+
+- Host
+- Host-page-locked
+- GPU
+
+Allocating and freeing the GPU and host page-locked (or pinned) memory require
+device synchronization. As a result, when possible, DALI avoids reallocating these kinds of memory.
+The buffers that are allocated with this storage type will only grow when the existing buffer is too
+small to accommodate the requested shape. This strategy reduces the number of total memory
+management operations and increases the processing speed when the memory requirements become stable
+and no more allocations are required.
+
+By contrast, ordinary host memory is relatively inexpensive to allocate and free. To reduce
+host memory consumption, the buffers might shrink when the new requested size is smaller than
+the fraction of the old size. This is called shrink threshold. It can be adjusted to a value
+between 0 (never shrink) and 1 (always shrink). The default is 0.9. The value can be controlled
+by the ``DALI_HOST_BUFFER_SHRINK_THRESHOLD`` environmental variable or be set in Python by
+using the `nvidia.dali.backend.SetHostBufferShrinkThreshold` function.
+
+During processing, DALI works on batches of samples. For the GPU and some CPU operators, each batch
```

Comment:
```suggestion
During processing, DALI works on batches of samples. For GPU and some CPU operators, each batch
```
again, GPU is used as an adjective here.

Reply:
Done